dbos/_dbos_config.py

```diff
@@ -71,12 +110,55 @@ class ConfigFile(TypedDict, total=False):
     """
 
     name: str
-    language: str
     runtimeConfig: RuntimeConfig
     database: DatabaseConfig
     telemetry: Optional[TelemetryConfig]
     env: Dict[str, str]
-    application: Dict[str, Any]
+
+
+def is_dbos_configfile(data: Union[ConfigFile, DBOSConfig]) -> bool:
+    return "name" in data and (
+        "runtimeConfig" in data
+        or "database" in data
+        or "env" in data
+        or "telemetry" in data
+    )
```

Comment:
we set `total=False` on `ConfigFile`. This has the elegant property of allowing an easy fallback to the new struct if a user provides a `ConfigFile` with only the name, and recognize the internal struct if any other of its fields are provided.